### ISSUE TITLE
Strange object interaction between Lua <--> JS

### ISSUE THREAD

**Issue by greatwolf** [daurnimator/lua.vm.js#6]
**TL;DR**: After getting access to jQuery object from javascript into lua, early operations performed on that object causes subsequent operations to fail with `!Unsupported!` string being returned instead of the actual desired output.

I encountered this strange problem while testing out issue #5. I have the following simple html with some embedded Lua using jQuery:

```
<html>
  <head>
    <title>script example</title>
  </head>

  <script src="lua.vm.js"></script>  
  <script src="jquery-2.0.3.js"></script>

  <script type="text/lua">
  local jq = js.get("$")

  local jqxhr = jq.get("./glossary.json")
  jqxhr.done(function() print(jqxhr.responseText) end)
  </script>
</html>
```

This parses the json file and prints it to `console.log` as expected. However, if I add some intermediate code, like getting jquery version, then things start failing in mysterious ways:

```
...
local jq = js.get("$")
local version = jq().jquery
print(version, '********')

local jqxhr = jq.get("./glossary.json")
jqxhr.done(function() print(jqxhr.responseText) end)
...
```

The `jq().jquery` call succeeds and the version gets printed but `jqxhr.done` fails:

```
ERROR: [string "input"]:14: attempt to call field 'done' (a string value)
```

In fact, `jqxhr.done` will contain the string '!Unsupported!'. Calling that like it's a function causes the above error.

If I switch the order of the statements to this:

```
...
local jq = js.get("$")
local jqxhr = jq.get("./glossary.json")
local version = jq().jquery

print(version, '********')
jqxhr.done(function() print(jqxhr.responseText) end)
...
```

Then `jqxhr.done` has the right object again and the json parse results get printed like before. However, now the call `jq().jquery` fails and `version` will contain "!Unsupported!" instead of the expected jQuery version string.

What could be causing this bizarre behavior? This can be reproduced with jQuery 1.10.x as well as jQuery 2.0.x.
